### DISCUSSION TITLE
How to set NotMapped Row property using stored procedure

### DISCUSSION THREAD

**Discussion by qubess** [serenity-is/Serenity#5364] [Help]
I want to set one of the NotMapped property in a Row class using other properties in same Row class. I cannot use Expression since I am loading Row from stored procedure and not from query.

I tried setting NotMapped property default value, not null but nothing seems working. This property gets completely ignored when getting the result.

**Comment by luongnv6** [comment]
After receiving data you need to set values for them (ex: foreach)

**Comment by wezmag** [comment]
In your `Repository.cs`, override `OnReturn` in `MyRetrieveHandler` and also `ProcessEntity` in `MyListHandler`.

``` csharp
private class MyRetrieveHandler : RetrieveRequestHandler<MyRow> {
    protected override void OnReturn()
    {
        Row.NotMappedField = defaultValue;
        base.OnReturn();
    }
}
private class MyListHandler : ListRequestHandler<MyRow> {
    protected override MyRow ProcessEntity(MyRow row)
    {
        row.NotMappedField = defaultValue;
        return base.ProcessEntity(row);
    }
}
```
